doc/chassis-http-server/chassis_http_server.md

```diff
@@ -0,0 +1,112 @@
+# Sonic Chassis http-server for internal image and configuration hosting #
+#### Rev 1.0
+
+## Table of Contents
+ * [Revision](#revision)
+ * [Scope](#scope)
+ * [Definition/Abbreviation](#definition)
+ * [Overview](#overview)
+ * [Requirements](#Requirements)
+ * [Architecture Design](#Architecture-Design)
+ * [High-Level Design](#High-Level-Design)
+ * [Warmboot and Fastboot Design Impact](#Warmboot-Fastboot-Design-Impact)
+ * [Unit Test](#Unit-Test-cases)
+
+### Revision  
+| Rev |     Date    |       Author       |        Change Description           |
+|:---:|:-----------:|:------------------:|:------------------------------------|
+| 1.0 | 03/15/2023  | Anand Mehra        | Initial Draft for http-server       |
+
+
+### Scope  
+
+ This document provides design details for the http-server for internal midplane in Sonic Chassis system.
+
+### Definitions/Abbreviations
+
+### Overview 
+
+ On a Sonic Chassis system, currently there is no support for hosting image and configuration files at a Centralized internal location, i.e., Supervisor, for sharing the same with the Line Card modules. Without this support, each Line Card modules must reach to external servers on the mgmt interface network to download these files.
+
+ The http-server implementation described in this document provides a way to run http-server on the supervisor to host image and configuration files for Line Card module and share those on the internal midplane network. This http-server implementation is not to service the requests on the external network.
+
+
+### Requirements 
+
+ The http-server shall run on the internal midplane network and use the internal Supervisor midplane IP, provided by the platform, to bind.
+ The http-server shall use port 8000 to bind on the internal network.
+
+ The use of this http-server shall be limited to internal midplane and not designed to service requests on the mgmt/external interface.
+
+### Architecture Design 
+
+ The new http-server shall use the current Sonic architecture and shall not require any architecture design change. The server shall run only on the supervisor and inside the existing database-chassis docker, which runs only on supervisor.
```

Comment:
Why put this service in database container? I do not see any dependency.

Reply:
It's in database-chassis container which runs only on Supervisor. Please see above comment for more details. Thanks